apps/deploy-web/src/components/deployments/DeploymentDetailTopBar.tsx

```diff
@@ -96,6 +104,34 @@ export const DeploymentDetailTopBar: React.FunctionComponent<Props> = ({ address
     }
   };
 
+  const setAutoTopUpEnabled = useCallback(
+    async (autoTopUpEnabled: boolean) => {
+      if (autoTopUpEnabled && realTimeLeft?.timeLeft) {
+        const secTillNextTopUp = differenceInSeconds(addHours(startOfHour(new Date()), 2), new Date());
+        const secTillClosed = differenceInSeconds(realTimeLeft.timeLeft, new Date());
+
+        if (secTillClosed < secTillNextTopUp) {
+          const secToDepositFor = secTillNextTopUp - secTillClosed;
+          const deposit = Math.ceil((deploymentCost * secToDepositFor) / 6);
+
+          const isConfirmed = await confirm({
+            title: "Deposit required",
+            message: `To enable auto top-up, you need to deposit $${udenomToUsd(deposit, deployment.escrowAccount.balance.denom)}. This is required to maintain your deployment til the next scheduled check.`
+          });
+
+          if (!isConfirmed) {
+            return;
+          }
+
+          await onDeploymentDeposit(deposit, browserEnvConfig.NEXT_PUBLIC_MASTER_WALLET_ADDRESS);
+        }
+      }
+
+      deploymentSetting.setAutoTopUpEnabled(autoTopUpEnabled);
```

Comment:
We should show a loading in the switch and disable it while this is being enabled, otherwise the delay feels weird when switching it on/off and it prevents spamming the control by accident.